index.d.ts

```diff
@@ -0,0 +1,50 @@
+type TimeWindow = number; // e.g. 60 * 1000
+
+interface Snapshot {
+    /**
+     * Reset the snapshot
+     */
+    reset();
+
+    /**
+     * 
+     * @param snapshot Append chunks of snapshot to the snapshot
+     */
+    add(snapshot: Snapshot);
+}
+
+export interface RollingWindowOptions {
+    numChunks?: number;
+    timeWindow?: TimeWindow;
+    buildHistogram?: () => Snapshot;
+}
+
+export class RollingWindow {
+    constructor(options?: RollingWindowOptions);
+
+    /**
+     * Rotate the rolling window positions
+     */
+    rotate();
+
+    /**
+     * Record value for the current window
+     */
+    recordValue(value: number);
+
+    /**
+     * Get the snapshot for the current window
+     * @param snapshot Optional snapshot if provided
+     */
+    getSnapshot(snapshot: Snapshot): Snapshot;
```

Comment:
This is something like?

```ts
// @param base Optional histogram to be used to aggregate histograms into if provided
getSnapshot(base?: Histogram): Histogram;
```

Reply:
I used `AbstractHistogram` from `hdr-histogram-js` instead to better reflect the exact type

Reply:
Ah, that's a good idea!